packages/react-components/src/components/xy-output-component.tsx

```diff
@@ -288,6 +291,191 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
+            this.updatePositionAfterPann(right,distance);
+            this.props.unitController.viewRange = {
+                start: this.props.unitController.absoluteRange - this.props.unitController.viewRangeLength,
+                end: this.props.unitController.absoluteRange
+            };
+        } else {
+            this.updatePositionAfterPann(right, percentRange);
+            this.props.unitController.viewRange = {
+                start: this.NewstartEndRangePanning(right, percentRange)[0],
+                end: this.NewstartEndRangePanning(right, percentRange)[1]
+            };
+        }
+    }
+
+    private shiftLeft() {
+        const numberLeft = -1;
+        const distance = ( this.props.unitController.viewRange.start);
+        this.shift(numberLeft, distance);
+    }
+
+    private shiftRight() {
+        const numberRight = 1;
+        const distance = (this.props.unitController.absoluteRange - this.props.unitController.viewRange.end);
+        this.shift(numberRight, distance);
+    }
+
+    private wheelEvent(wheel: React.WheelEvent) {
+        if (wheel.shiftKey) {
+            if (wheel.deltaY < 0) {
+                this.shiftLeft();
+            }
+            else if (wheel.deltaY > 0) {
+                this.shiftRight();
+            }
+        }
+    }
+
+    private getPosition(event: React.MouseEvent): number {
+        const offset = this.props.viewRange.getOffset() ?? 0;
+        const scale = this.props.viewRange.getEnd() - this.props.viewRange.getstart();
+        const xPos = this.props.viewRange.getstart() - offset +
+            (event.nativeEvent.offsetX / this.lineChartRef.current.chartInstance.width) * scale;
+        return xPos;
+    }
+
+    private MoveEvent(event: React.MouseEvent) {
+        if (!this.mouseIsDown) {
+            this.positionZoomMouse = this.getPosition(event);
+        }
+    }
+
+    private initializePosition(): number[] {
+        let position = 0;
+        if (this.isZoomCenter) {
+            this.positionZoomCenter = (this.props.unitController.viewRange.end + this.props.unitController.viewRange.start) / 2;
+            position = this.positionZoomCenter;
+        } else {
+            position = this.positionZoomMouse;
+        }
+        const startPosition = position - this.props.unitController.viewRange.start;
+        const endPosition = this.props.unitController.viewRange.end - position;
+        return [startPosition, endPosition];
+    }
+
+    private mousePositionPercent(): number {
+        const diffStartMouse = this.positionZoomMouse - this.props.unitController.viewRange.start;
+        const percentMousePosition = diffStartMouse / this.props.unitController.viewRangeLength;
+        return percentMousePosition;
+    }
+
+    private zoomKey(key: React.KeyboardEvent) {
+        switch (key.key) {
+            case '+':
+            case '=': {
+                this.isZoomCenter = true;
```

Comment:
Not a good idea to set an instance variable so that it can be used by a method you call, it should just be a method parameter.